### PR TITLE
Replace nose with pytest, v2

### PR DESCRIPTION
**Pull request by hrichards**
@ctk3b 

Extends (and thus replaces) @ctk3b's original PyTest PR (https://github.com/counsyl/capone/pull/7): it still upgrades to PyTest, but it cleans up and removes a lot of dependencies that caused the project to no longer build or the tests to no longer run.

I've dropped:

1.  Python 2 support
2.  `nose-parameterized`
3.  `pytest-xdist` (for small projects, I've always found the overhead for `pytest-xdist` to be more trouble than it's worth)
4.  Python 2-isms like `unicode_literals`

But the bones of this PR are still mostly yours, @ctk3b.  Thanks again!

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 language: python
 python:
   - "2.7"
```
Comment (ctk3b) [inline .travis.yml:3]: Do we want to drop testing for py2 as well and add py37?

File: capone/tests/test_filter_by_related_objects.py [hunk 2]
```diff
@@ -1,8 +1,7 @@
-from __future__ import unicode_literals
 from decimal import Decimal as D
 
 from django.test import TestCase
-from nose_parameterized import parameterized
+from parameterized import parameterized
```
Comment (ctk3b) [inline capone/tests/test_filter_by_related_objects.py:4]: I think we can punt on this but should be using pytest's version in the future.

Comment (hrichards) [reply capone/tests/test_filter_by_related_objects.py:4]: I tried it, but it doesn't work on Django's TestCase methods.

File: .travis.yml (hunk 1 above)
Comment (hrichards) [reply .travis.yml:3]: Good point.  Missed that reference.  See a38a4cbbb2149690ff4b80cb0da1bbfd5e713806.

**Comment by hrichards** [comment]
@ctk3b Do you know why Travis isn't running here?

**Comment by hrichards** [comment]
Looks like the jobs are running, but not being reported back to here.  There are also a bunch of errors :-(

https://travis-ci.org/github/counsyl/capone/builds/670244206?utm_medium=notification&utm_source=github_status

**Comment by hrichards** [comment]
So, the little green check mark isn't showing up here for some reason, but the build was triggered properly and has succeeded!

https://travis-ci.org/github/counsyl/capone/builds/670337225?utm_medium=notification&utm_source=github_status

Turns out I was on the right track: I *did* need to declare that I needed Postgres to run these tests: the solution was found here:

https://travis-ci.org/github/counsyl/capone/builds/670337225?utm_medium=notification&utm_source=github_status

and the needed lines (in f2c45d6) were as follows:

```
services:
  - postgresql
```

**Comment by ctk3b** [comment]
LGTM

Confirmed that the two parent hashes here are the ones we wanted to be testing: https://github.com/counsyl/capone/commit/747c3cb01fa5147d2907e0d6ad40df3f19315459

**Comment by hrichards** [comment]
Thanks, @ctk3b!  I bumped the version and will now merge.  IIRC, that will trigger our internal pipeline and promote this to PyPI.

**Pull request merged**